File: trabajo 10/metodo_quick sort.cs
using System;

class Program
{
    // Método Quick Sort
    static void QuickSort(int[] arr, int inicio, int fin)
    {
        if (inicio < fin)
        {
            int indicePivote = Particion(arr, inicio, fin);
            QuickSort(arr, inicio, indicePivote - 1);
            QuickSort(arr, indicePivote + 1, fin);
        }
    }

    static int Particion(int[] arr, int inicio, int fin)
    {
        int pivote = arr[fin];
        int i = inicio - 1;

        for (int j = inicio; j < fin; j++)
        {
            if (arr[j] <= pivote)
            {
                i++;
                Intercambiar(arr, i, j);
            }
        }
        Intercambiar(arr, i + 1, fin);
        return i + 1;
    }

    static void Intercambiar(int[] arr, int a, int b)
    {
        int temp = arr[a];
        arr[a] = arr[b];
        arr[b] = temp;
    }

    static void ImprimirArreglo(int[] arr)
    {
        foreach (int num in arr)
        {
            Console.Write(num + " ");
        }
        Console.WriteLine();
    }

    static void Main()
    {
        Random rnd = new Random();
        int tamaño = 10;
        int[] lista = new int[tamaño];

        // Rellenar con números aleatorios entre 1 y 100
        for (int i = 0; i < tamaño; i++)
        {
            lista[i] = rnd.Next(1, 101);
        }

        Console.WriteLine("Lista desordenada:");
        ImprimirArreglo(lista);

        QuickSort(lista, 0, tamaño - 1);

        Console.WriteLine("\nLista ordenada:");
        ImprimirArreglo(lista);
    }
}

File: trabajo 10/metodo_quick sort.js
// Función para intercambiar elementos en el arreglo
function intercambiar(arr, a, b) {
    let temp = arr[a];
    arr[a] = arr[b];
    arr[b] = temp;
}

// Función de partición para Quick Sort
function particion(arr, inicio, fin) {
    let pivote = arr[fin];
    let i = inicio - 1;

    for (let j = inicio; j < fin; j++) {
        if (arr[j] <= pivote) {
            i++;
            intercambiar(arr, i, j);
        }
    }
    intercambiar(arr, i + 1, fin);
    return i + 1;
}

// Función Quick Sort recursiva
function quickSort(arr, inicio, fin) {
    if (inicio < fin) {
        let indicePivote = particion(arr, inicio, fin);
        quickSort(arr, inicio, indicePivote - 1);
        quickSort(arr, indicePivote + 1, fin);
    }
}


function imprimirArreglo(arr) {
    console.log(arr.join(' '));
}



// Generar arreglo con números aleatorios entre 1 y 100
const tamaño = 10;
let lista = [];
for (let i = 0; i < tamaño; i++) {
    lista.push(Math.floor(Math.random() * 100) + 1);
}

console.log("Lista desordenada:");
imprimirArreglo(lista);

quickSort(lista, 0, lista.length - 1);

console.log("\nLista ordenada:");
imprimirArreglo(lista);

